var/spack/repos/builtin/packages/py-hail/package.py

```diff
@@ -0,0 +1,236 @@
+        depends_on("py-pyyaml@6.0:7")
+        depends_on("py-sortedcontainers@2.4.0:2")
+        depends_on("py-tabulate@0.8.9:0")
+        depends_on("py-uvloop@0.19.0:0")
+        depends_on("py-jproperties@2.1.1:2")
+        # Undocumented runtime requirements for hailtop
+        # These are also required to use the HAIL API
+        # but are not explicitly mentioned anywhere
+        depends_on("py-azure-mgmt-core")
+        depends_on("py-typing-extensions")
+
+    patch("fix-lz4-import-builtins.patch")
+
+    build_directory = "hail"
+
+    def patch(self):
+        # Hail will fail to build if it cannot determine a commit hash from git
+        # which will not be available in a spack cache. Since we know it from
+        # the package, we can inject it in the failure and move forward.
+        revision = self.hail_revision
+
+        filter_file(
+            r'\$\(error "git rev-parse HEAD" failed to produce output\)',
+            f"REVISION := {revision}",
+            "hail/version.mk",
+        )
+        filter_file(
+            r'\$\(error "git rev-parse --short=12 HEAD" failed to produce output\)',
+            f"SHORT_REVISION := {revision[:12]}",
+            "hail/version.mk",
+        )
+        filter_file(
+            r'\$\(error "git rev-parse --abbrev-ref HEAD" failed to produce output\)',
+            f"BRANCH := tags/{version}",
+            "hail/version.mk",
+        )
+
+        # Also need to make sure that build-info.properties gets the right revision
+        # which ends up improperly calculated in scala and will crash at runtime
+        filter_file(
+            r"val revision = VcsVersion\.vcsState\(\)\.currentRevision",
+            "val vcs_revision = VcsVersion.vcsState().currentRevision\n"
+            f'  val revision = if(vcs_revision ==  "no-vcs") "{revision}" else vcs_revision\n',
+            "hail/build.sc",
+        )
+
+    @property
+    def hail_revision(self):
+        version = self.version
+        version_info = self.versions[version]
+        # REVISION must look like a hash or Hail crashes at startup
+        # Technically, it needs to be at least 12 characters
+        revision = version_info.get("commit", version.joined.string.ljust(40, "0"))
+        return revision
+
+    @property
+    def hail_pip_version(self):
+        # This is the same behavior is as is defined in hail/version.mk
+        return f"{self.spec.version.up_to(3)}"
+
+    @property
+    def build_wheel_file_path(self):
+        wheel_file = f"hail-{self.hail_pip_version}-py3-none-any.whl"
+        wheel_dir = join_path("build", "deploy", "dist")
+        return join_path(wheel_dir, wheel_file)
+
+    def setup_build_environment(self, env):
+        # HAIL build doesn't find lz4: https://discuss.hail.is/t/ld-pruning-repeated-errors/1838/14
+        env.append_flags("CXXFLAGS", f"-I{self.spec['lz4'].prefix.include}")
+
+    def setup_run_environment(self, env):
+        pass
```

Comment:
:thinking:

Reply:
There's probably some environmental variables that can be set for optimal performance. Not required but perhaps for a future iteration. Will remove for now.